### PR TITLE
Forces Drone to Stay on the Station

### PR DESCRIPTION
**Pull request by JamieD1**
Forces Drone to Stay on the Station

:cl:  
rscadd: Forces Drone to Stay on the Station
/:cl:

### CODE REVIEW

**Comment by Gboster-0** [comment]
I feel like this should be more enforced gameplay wise (for example marking what z-level the drone was spawned at and exploding it if it goes on any other z-level)

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 1]
```diff
@@ -62,7 +62,7 @@
 	var/laws = {"\
 1. You may not involve yourself in the matters of another being, even if such matters conflict with Law Two or Law Three, unless the other being is another Drone.
 2. You may not harm any being, regardless of intent or circumstance.
-3. Your goals are to build, maintain, repair, improve, and provide power to the best of your abilities, You must never actively work against these goals.\
+3. Your goals are to build, maintain, repair, improve, and provide power of your assigned station to the best of your abilities, You must never actively work against these goals or leave your assigned station.\
```
Comment (ReddicusDragon) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]:
```suggestion
3. Your goals are to build, maintain, repair, improve, and provide power to your assigned station to the best of your abilities, You must never actively work against these goals or leave your assigned station.\
```
grammar go brrrt

**Comment by Sniblet** [comment]
ok but why?

**Comment by Moltijoe** [comment]
> I feel like this should be more enforced gameplay wise (for example marking what z-level the drone was spawned at and exploding it if it goes on any other z-level)

not ideal since they might end up accidentally spacing themselves, dying, they should just be warped randomly station side

**Comment by Burger1998** [comment]
what if they need to fix mining base or gulag? drone has a few more zlevels they need to be then other locked ghost roles such as space bartender

**Comment by CurrTheLost** [comment]
I'd say this is a bad one.

**Review by Moltijoe — APPROVED** [review]
Since this is just a text change and not actually a hard limit, i think it's fair

**Comment by Sniblet** [comment]
It’s a law change. It’s a hard limit.
Has anyone explained how this improves the game? Does the head dev just get to do this kind of thing with no support or communication? How is this not missing rationale?

**Comment by Cartlord** [comment]
Bibby did actually label it as missing rationale, Jamie removed those labels because he felt like it.

**Comment by CurrTheLost** [comment]
Nobody is liking this, Jamie. No rationale, no reasoning, arbitrary placement of a rule, and can come as a complete and total surprise to people with severe complaints from people who have it happen to them...

**Comment by CurrTheLost** [comment]
> what if they need to fix mining base or gulag? drone has a few more zlevels they need to be then other locked ghost roles such as space bartender

This, mainly due to bardrone, by the way.

**Comment by ReddicusDragon** [comment]
@JamieD1 to assuage everyone's concerns, maybe change the wording to "your assigned station and it's mining outpost"

Mining base feels weird to put but that'd work too

**Pull request merged**